sdk/dotnet/Pulumi/Core/Id.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright 2016-2018, Pulumi Corporation
+
+#nullable enable
+
+using System;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// A provider-assigned ID.
+    /// </summary>
+    public readonly struct Id : IEquatable<Id>

```

Comment:
Should we consider overloading `operator==` and `operator!=`?

Reply:
they are?